lib/extensions/conditional/active_record/base.rb

```diff
@@ -54,6 +54,16 @@ def dependent_object
     def satisfied_by?(_user)
       raise NotImplementedError
     end
+
+    private
+
+    # @return [Boolean] true if the current condition result in a cyclic dependency.
```

Comment:
I think this definition needs to be refined.

If an edge is added to a DAG that makes it cyclic, then more than one edge causes the DAG to be cyclic.